src/toil/test/jobStores/jobStoreTest.py

```diff
@@ -456,17 +459,35 @@ def testImportSharedFile(self, otherCls):
                        otherCls=activeTestClassesByName)
 
         def testImportHttpFile(self):
-            srcUrl, srcHash = ('https://raw.githubusercontent.com/BD2KGenomics/toil'
-                               '/12c711ff91caa5d8fb37a072bad376f447faa797/Makefile',
-                               '16a38fd7ae233616083d43d81159d4e8')
-            with self.master.readFileStream(self.master.importFile(srcUrl)) as f:
-                self.assertEqual(hashlib.md5(f.read()).hexdigest(), srcHash)
+            http = SocketServer.TCPServer(('', 0), StubHttpRequestHandler)
+            try:
+                httpThread = threading.Thread(target=http.serve_forever)
+                httpThread.start()
+                try:
+                    assignedPort = http.server_address[1]
+                    url = 'http://localhost:%d' % assignedPort
+                    with self.master.readFileStream(self.master.importFile(url)) as readable:
+                        self.assertEqual(readable.read(), StubHttpRequestHandler.fileContents)
+                finally:
+                    try:
+                        http.shutdown()
```

Comment:
@hannes-ucsc shutdown is idempotent and must be called before the thread is joined.